scripts/import_linter/import_linter.py

```diff
@@ -0,0 +1,53 @@
+import os
+import subprocess
+import multiprocessing
+import importlib
+
+git_base = subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode().strip()
+
+
+def walk_and_ignore_pycache(directory):
+    list = []
+    for root, dirnames, files in os.walk(directory, topdown=True):
+        # This line removes any __pycache__ directories from the list
+        dirnames[:] = [d for d in dirnames if d != '__pycache__' and d != 'tests' and d != 'data']
+        filenames = [f for f in files if f.endswith('.py') and not f.startswith('__init__')]
+        for filename in filenames:
+            # Process files as you would like
+            list.append(os.path.join(root, filename))
+    return list
+
+
+def file_to_import(file):
+    push_file = []
+    head_tail = os.path.split(file)
+    while head_tail[1] != "promptflow" and head_tail[0] != "":
+        if head_tail[1].endswith(".py"):
+            push_file.insert(0, head_tail[1][:-3])
+        else:
+            push_file.insert(0, head_tail[1])
+        file = head_tail[0]
+        head_tail = os.path.split(file)
+    push_file.insert(0, "promptflow")
+    return ".".join(push_file)
+
+
+def subprocess_check_python_import(file):
+    print(f'Checking import of {file} on process ID: {os.getpid()}')
+    importlib.import_module(file)
```

Comment:
may add a comment to demonstrate what does `importlib.import_module` consume? and what will happen if it finds some improper import?

Reply:
OK